sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1054 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_ieee802154_hal IEEE802.15.4 Radio Hardware Abstraction Layer
+ * @ingroup     drivers
+ * @brief       This is a Hardware Abstraction Layer for IEEE802.15.4 compatible
+ *              radios.
+ *
+ * @author      José I. Alamos <jose.alamos@haw-hamburg.de>
+ */
+
+#ifndef IEEE802154_RADIO_H
+#define IEEE802154_RADIO_H
+
+#include <stdbool.h>
+#include "iolist.h"
+#include "sys/uio.h"
+#include "byteorder.h"
+#include "net/eui64.h"
+
+/**
+ * @brief Forward declaration of the radio ops structure.
+ */
+typedef struct ieee802154_radio_ops ieee802154_radio_ops_t;
+
+typedef enum {
+    /**
+     * @brief the transceiver successfully sent a frame.
+     *
+     * Depending of the type of transmissions and available caps, this could mean one of the following:
+     * If the device supports @ref IEEE802154_CAP_FRAME_RETRIES or @ref IEEE802154_CAP_IRQ_ACK_TIMEOUT:
+     * - The frame was sent without ACK Req bit
+     * - The frame was sent with the ACK bit and a valid ACK was received.
+     *
+     * Otherwise, this notifies that a frame was sent.
+     */
+    TX_STATUS_SUCCESS,
+    /**
+     * @brief the transceiver received a valid ACK with the frame pending bit
+     *
+     * This status is present only if the device supports @ref
+     * IEEE802154_CAP_FRAME_RETRIES or @ref IEEE802154_CAP_IRQ_ACK_TIMEOUT.
+     */
+    TX_STATUS_FRAME_PENDING,
+    /**
+     * @brief the transceiver ran out of retransmission
+     *
+     * This status is present only if the device supports @ref
+     * IEEE802154_CAP_FRAME_RETRIES or @ref IEEE802154_CAP_IRQ_ACK_TIMEOUT.
+     */
+    TX_STATUS_NO_ACK,
+    /**
+     * @brief the CSMA-CA algorithm or CCA failed to measure a clear channel
+     */
+    TX_STATUS_MEDIUM_BUSY,
+} ieee802154_tx_status_t;
+
+/**
+ * @brief IEEE802.15.4 Radio capabilities
+ *
+ * These flags represent the hardware capabilities of a given device.
+ */
+typedef enum {
+    /**
+     * @brief the device supports frame retransmissions with CSMA-CA
+     *
+     * @note it's implicit that a radio supports @ref IEEE802154_CAP_AUTO_CSMA
+     * if this cap is available
+     */
+    IEEE802154_CAP_FRAME_RETRIES,
+    /**
+     * @brief the device supports Auto CSMA-CA
```

Comment:
Please add a short description what that means / what is required for a driver to expose that capability.

Reply:
done